internal/provider/event_collectors/install/splunk/audit_logs.go

```diff
@@ -0,0 +1,297 @@
+	}
+
+	return &data
+}
+
+func (r *AuditLogs) toJson(data any) any {
+	json := auditLogsJsonReadWrite{}
+	datav, ok := data.(*auditLogsModel)
+	if !ok {
+		return nil
+	}
+
+	if !datav.HecEndpoint.IsNull() && !datav.HecEndpoint.IsUnknown() {
+		hecEndpoint := datav.HecEndpoint.ValueString()
+		json.HecEndpoint = &hecEndpoint
+	}
+
+	token := TokenReadWrite{}
+	if !datav.HecTokenClearText.IsNull() && !datav.HecTokenClearText.IsUnknown() {
+		hecToken := datav.HecTokenClearText.ValueString()
+		token.ClearText = &hecToken
+		json.HecToken = &token
+	}
+
+	if !datav.HecTokenHash.IsNull() && !datav.HecTokenHash.IsUnknown() {
+		hecToken := datav.HecTokenHash.ValueString()
+		token.Hash = &hecToken
+		json.HecToken = &token
+	}
+
+	return json
+}
+
+func (r *AuditLogs) getId(data any) *string {
+	model, ok := data.(*auditLogsModel)
+	if !ok {
+		return nil
+	}
+
+	str := model.Token.ValueString()
+	return &str
+}
+
+func (r *AuditLogs) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	providerData := internal.Configure(&req, resp)
+
+	r.installer = &common.Install{
+		Integration:  SplunkIntegration,
+		Component:    AuditLogsComponent,
+		ProviderData: providerData,
+		GetId:        r.getId,
+		GetItemJson:  r.getItemJson,
+		FromJson:     r.fromJson,
+		ToJson:       r.toJson,
+	}
+}
+
+func (s *AuditLogs) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	var plan auditLogsModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	// Generate a UUID identifier for the token
+	u, err := uuid.GenerateUUID()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"UUID Generation Error",
+			"Failed to generate a UUID for the token.",
+		)
+		return
+	}
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Ensure the cleartext field is provided
+	if plan.HecTokenClearText.IsNull() || plan.HecTokenClearText.ValueString() == "" {
+		resp.Diagnostics.AddError(
+			"Missing Required Attribute",
+			"The 'hec_token_cleartext' field is required for resource creation.",
+		)
+		return
+	}
+
+	var api auditLogsApiReadWrite
+	var model auditLogsModel
+
+	plan.Token = types.StringValue(u)
+	resp.Diagnostics.Append(req.Plan.SetAttribute(ctx, path.Root("token"), types.StringValue(u))...)
+
+	s.installer.EnsureConfig(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &model)
+	s.installer.Stage(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &api, &model)
+	s.installer.UpsertFromStage(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &api, &model)
+
+	var currentTokenId types.String
+	resp.Diagnostics.Append(resp.State.GetAttribute(ctx, path.Root("token"), &currentTokenId)...)
+	if currentTokenId.ValueString() != "" {
+		// manually set the token cleartext attribute
+		// this is needed because the cleartext token is not returned by the API
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root(HecTokenClearTextKey), plan.HecTokenClearText)...)
```

Comment:
This value needs to be explicitly put back into the statefile because it is not returned by the API